src/lib/routing/router.ts

```diff
@@ -20,21 +20,47 @@ const defaultErrorHandler = (auto405) => async (err: Error, req: EReq, res: ERes
   res.withStatus(500).json({ error: err.message });
 }
 
+const preflightHandler = (autoCorsPreflight: AutoCorsPreflightOptions) => async (req: EReq, res: ERes) => {
+  if(autoCorsPreflight.trustedOrigins.length===0){
+    return;
+  }
+  let originHeaderValue :string | null = null;
+  if(autoCorsPreflight.trustedOrigins.length===1 && autoCorsPreflight.trustedOrigins[0] === '*'){
+    originHeaderValue = '*';
+  }else{
+    const origin = req.headers.get("origin");
+    if (autoCorsPreflight.trustedOrigins.includes(origin)){
+      originHeaderValue = origin;
+    }
+  }
+  if(originHeaderValue==null){
+    return;
+  }
+  res.set({
+    'access-control-allow-origin': originHeaderValue,
+    'access-control-allow-methods': 'GET, PUT, POST, DELETE, PATCH, OPTIONS'
```

Comment:
Here you'll want to echo back to the user the values they sent in the `Access-Control-Request-Method` and `Access-Control-Request-Header` headers, rather than heard coding the values